go/vt/vtadmin/api_test.go

```diff
@@ -63,6 +59,11 @@ import (
 	"vitess.io/vitess/go/vt/proto/vttime"
 )
 
+func TestMain(m *testing.M) {
```

Comment:
I don't follow. Why can't we use `go test -run` ?

Reply:
On `main` it doesn't seem to be parse the test flags by default. I checked on `release13.0`, just so that I went back a bit and it works fine.

On `release13.0`
```
rohit@rohit-ubuntu:~/vitess-ps/go/vt/vtadmin$ go test -v -run TestGetGates
=== RUN   TestGetGates
=== PAUSE TestGetGates
=== CONT  TestGetGates
--- PASS: TestGetGates (0.00s)
PASS
ok  	vitess.io/vitess/go/vt/vtadmin	0.013s
```

On `main`
```
rohit@rohit-ubuntu:~/vitess-ps/go/vt/vtadmin$ go test -v -run TestGetGates
W0904 12:20:08.309102   29964 cache.go:119] BackfillEnqueueWaitTime (0s) must be positive, defaulting to 50ms
W0904 12:20:08.309322   29964 cache.go:124] BackfillRequestTTL (0s) must be positive, defaulting to 100ms
W0904 12:20:08.309456   29964 cache.go:119] BackfillEnqueueWaitTime (0s) must be positive, defaulting to 50ms
...
W0904 12:20:08.352305   29964 cache.go:124] BackfillRequestTTL (0s) must be positive, defaulting to 100ms
W0904 12:20:08.352516   29964 cache.go:119] BackfillEnqueueWaitTime (0s) must be positive, defaulting to 50ms
W0904 12:20:08.352524   29964 cache.go:124] BackfillRequestTTL (0s) must be positive, defaulting to 100ms
PASS
ok  	vitess.io/vitess/go/vt/vtadmin	0.071s
```

I added logs inside the go flag/testing modules and the issue seems to be that the `init()` in `test_tmclient.go` parses the command line without the test flags and then when the default `go test` main runs it finds that the flags are already parsed and hence the test flags never get parsed.